src/LogicBlock/SMTLibLogicModel.cpp

```diff
@@ -0,0 +1,131 @@
+            if (m.size() == 4) {
+                if (m[1] == a.getName()) {
+                    return line;
+                }
+            }
+        }
+    }
+    return {};
+}
+std::string smtliblogic::SMTLibLogicModel::getTermValue(const logicbase::LogicTerm& a) {
+    std::string const termString = getTermString(a);
+
+    const auto  regex = std::regex(R"(\(define-fun (.*) \(\) (.*) (.*)\))");
+    std::smatch m;
+
+    if (std::regex_search(termString, m, regex)) {
+        if (m.size() == 4) {
+            return m[3];
+        }
+    }
+
+    return {};
+}
+int smtliblogic::SMTLibLogicModel::getIntValue(const logicbase::LogicTerm& a, logicbase::LogicBlock* lb) {
+    std::string const value = getTermValue(a);
+    const auto        type  = getTermType(a);
+    if (type != "Int") {
+        throw std::runtime_error("SMTLibLogicModel::getIntValue: type mismatch");
+    }
+    if (value.empty()) {
+        return 0;
+    }
+    return std::stoi(value);
+}
+bool smtliblogic::SMTLibLogicModel::getBoolValue(const logicbase::LogicTerm& a, logicbase::LogicBlock* lb) {
+    std::string const value = getTermValue(a);
+    const auto        type  = getTermType(a);
+    if (type != "Bool") {
+        throw std::runtime_error("SMTLibLogicModel::getBoolValue: type mismatch");
+    }
+    if (value.empty()) {
+        return false;
+    }
+    return value == "true";
+}
+double smtliblogic::SMTLibLogicModel::getRealValue(const logicbase::LogicTerm& a, logicbase::LogicBlock* lb) {
+    std::string const value = getTermValue(a);
+    const auto        type  = getTermType(a);
+    if (type != "Real") {
+        throw std::runtime_error("SMTLibLogicModel::getRealValue: type mismatch");
+    }
+    if (value.empty()) {
+        return 0.0;
+    }
+    return std::stod(value);
+}
+uint64_t smtliblogic::SMTLibLogicModel::getBitvectorValue(const logicbase::LogicTerm& a, logicbase::LogicBlock* lb) {
+    std::string const value = getTermValue(a);
+    if (value.empty()) {
+        return 0;
+    }
+    const auto type = getTermType(a);
+
+    const auto  typeRegex = std::regex(R"(\(_ BitVector (\d+)\))");
+    std::smatch m;
+    if (std::regex_search(type, m, typeRegex)) {
+        if (m.size() == 2) {
+            if (std::stoi(m[1]) != a.getBitVectorSize()) {
+                throw std::runtime_error("SMTLibLogicModel::getBitvectorValue: bitvector size mismatch");
+            }
+        } else {
+            throw std::runtime_error("SMTLibLogicModel::getBitvectorValue: bitvector size not found");
+        }
+    } else {
+        throw std::runtime_error("SMTLibLogicModel::getBitvectorValue: bitvector type not found");
+    }
+    const auto regex = std::regex(R"(#x(\d+))");
+
+    if (std::regex_search(value, m, regex)) {
+        if (m.size() == 2) {
+            return std::stoull(m[1], 0, 2);
+        }
+    }
+    return 0;
+}
+logicbase::LogicTerm smtliblogic::SMTLibLogicModel::getValue(const logicbase::LogicTerm& a, logicbase::LogicBlock* lb) {
```

Comment:
Maybe I am just wrong, but I do not see the `logicbase::LogicBlock* lb` being used in any of the functions and it's always `nullptr` in the tests. Might as well remove it then.

Reply:
they are there for compatability with other Model classes, for example the z3-Model, to have a drop in replacement

Reply:
Ok. that makes sense. Consider marking them `[[maybe_unused]]` to make this a little more explicit.